### ISSUE TITLE
Support P2WPKH to single-sig P2TR wallet

### ISSUE THREAD

**Issue by notmandatory** [bitcoindevkit/bdk#394]
As suggested by @moneyball on Discord, this feature could be a transitional step to full TR adoption and valuable to projects building single signature P2WPKH wallets. The idea is described in more detail in bullet 4 of [Bitcoin Optech "Preparing for taproot"](https://bitcoinops.org/en/preparing-for-taproot/).

The way I believe this would work for BDK is the user would create a wallet with a wpkh() descriptor, and then have an option to use this wallet to derive new P2TR addresses, scan the blockchain for transactions with their P2TR addresses, and create and sign transactions that spend their P2TR outputs. All other BDK wallet features would also need to be updated such as for determining balances, and listing wallet transaction and unspent utxos.

**Comment by afilini** [comment]
I think this could be a bit out of scope for BDK but maybe we can provide examples to help our users do that themselves. I think this connects back to the discussion we had a while back about having wallets with multiple descriptors: doing it internally basically means having a special case for the `wpkh` descriptor and offering the user the option to treat the wallet as a multi-descriptor wallet, with the original `wpkh` and the new `tr` one.

I think it would overcomplicate things a bit.

**Comment by moneyball** [comment]
LDK has the notion of batteries/modules. The core API is highly flexible
and does not assume much. But LDK batteries are specific implementations
that can also be reusable libraries. Would it makes sense for BDK to offer
similar "batteries"? Or perhaps think of it as a layer on top of BDK that
implements common patterns for wallet developers?

**Comment by afilini** [comment]
Yeah conceptually I'd like to have something similar. It could be both, maybe a component that works with BDK as a module or sometimes some kind of wrapper on our wallet.

But I like the idea of providing bits of code that people can take and adapt for their use-case.

**Comment by notmandatory** [comment]
Track via #63

**Comment by notmandatory** [comment]
Will require #486
